### PR TITLE
DBZ-6209 Refactor storage implementations

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-6209

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/history/AbstractFileSchemaHistory.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.util.FunctionalReadWriteLock;
+import io.debezium.util.Loggings;
+
+public abstract class AbstractFileSchemaHistory extends AbstractSchemaHistory {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/relational/history/AbstractFileSchemaHistory.java:28]:
```suggestion
public abstract class AbstractFileBasedSchemaHistory extends AbstractSchemaHistory {
```

File: debezium-storage/debezium-storage-azure-blob/src/main/java/io/debezium/storage/azure/blob/history/AzureBlobSchemaHistory.java [hunk 2]
```diff
@@ -71,46 +57,32 @@ public class AzureBlobSchemaHistory extends AbstractSchemaHistory {
...
-    private volatile List<HistoryRecord> records = new ArrayList<>();
+    private String container;
+    private String blobName;
 
     @Override
     public void configure(Configuration config, HistoryRecordComparator comparator, SchemaHistoryListener listener, boolean useCatalogBeforeSchema) {
         super.configure(config, comparator, listener, useCatalogBeforeSchema);
         if (!config.validateAndRecord(ALL_FIELDS, LOGGER::error)) {
-            throw new SchemaHistoryException("Error configuring an instance of " + getClass().getSimpleName() + "; check the logs for details");
+            throw new ConnectException("Error configuring an instance of " + getClass().getSimpleName() + "; check the logs for details");
```
Comment (jpechane) [inline debezium-storage/debezium-storage-azure-blob/src/main/java/io/debezium/storage/azure/blob/history/AzureBlobSchemaHistory.java:81]: Can we use `DebeziumException` here and in other places too?

**Review by jpechane — CHANGES_REQUESTED** [review]
@harveyyue Thanks lot. The PR looks good, I left just few minor comments. Unfortuntely the CI is failing. Could you please take a look?

File: debezium-storage/debezium-storage-azure-blob/src/main/java/io/debezium/storage/azure/blob/history/AzureBlobSchemaHistory.java (hunk 2 above)
Comment (harveyyue) [reply debezium-storage/debezium-storage-azure-blob/src/main/java/io/debezium/storage/azure/blob/history/AzureBlobSchemaHistory.java]: Done.

File: debezium-core/src/main/java/io/debezium/relational/history/AbstractFileSchemaHistory.java (hunk 1 above)
Comment (harveyyue) [reply debezium-core/src/main/java/io/debezium/relational/history/AbstractFileSchemaHistory.java]: Done.

**Comment by jpechane** [comment]
@harveyyue Applied, thanks!

**Pull request merged**